contracts/sol6/Dao/mock/KyberTokenFeeHandler.sol

```diff
@@ -0,0 +1,582 @@
+    /// @return amountTwei amount of rebate claimed
+    function claimReserveRebate(address rebateWallet) 
+        external 
+        override 
+        nonReentrant 
+        returns (uint256 amountTwei) 
+    {
+        require(rebatePerWallet[rebateWallet] > 1, "no rebate to claim");
+        // Get total amount of rebate accumulated
+        amountTwei = rebatePerWallet[rebateWallet].sub(1);
+
+        // redundant check, can't happen
+        assert(totalPayoutBalance >= amountTwei);
+        totalPayoutBalance = totalPayoutBalance.sub(amountTwei);
+
+        rebatePerWallet[rebateWallet] = 1; // avoid zero to non zero storage cost
+
+        // send rebate to rebate wallet
+        quoteToken.safeTransfer(rebateWallet, amountTwei);
+
+        emit RebatePaid(rebateWallet, quoteToken, amountTwei);
+
+        return amountTwei;
+    }
+
+    /// @dev claim accumulated fee per platform wallet. Called by any address
+    /// @param platformWallet the wallet to claim fee for. Total accumulated fee sent to this wallet.
+    /// @return amountTwei amount of fee claimed
+    function claimPlatformFee(address platformWallet)
+        external
+        override
+        nonReentrant
+        returns (uint256 amountTwei)
+    {
+        require(feePerPlatformWallet[platformWallet] > 1, "no fee to claim");
+        // Get total amount of fees accumulated
+        amountTwei = feePerPlatformWallet[platformWallet].sub(1);
+
+        // redundant check, can't happen
+        assert(totalPayoutBalance >= amountTwei);
+        totalPayoutBalance = totalPayoutBalance.sub(amountTwei);
+
+        feePerPlatformWallet[platformWallet] = 1; // avoid zero to non zero storage cost
+
+        quoteToken.safeTransfer(platformWallet, amountTwei);
+
+        emit PlatformFeePaid(platformWallet, quoteToken, amountTwei);
+        return amountTwei;
+    }
+
+    /// @dev set kyberDao contract address once and set setter address to zero.
+    /// @param _kyberDao kyberDao address.
+    function setDaoContract(IKyberDao _kyberDao) external {
+        require(msg.sender == daoSetter, "only daoSetter");
+        require(_kyberDao != IKyberDao(0));
+        kyberDao = _kyberDao;
+        emit KyberDaoAddressSet(kyberDao);
+
+        daoSetter = address(0);
+    }
+
+    /// @dev set new kyberNetwork address by daoOperator
+    /// @param _kyberNetwork new kyberNetwork contract
+    function setNetworkContract(address _kyberNetwork) external onlyDaoOperator {
+        require(_kyberNetwork != address(0), "kyberNetwork 0");
+        if (_kyberNetwork != kyberNetwork) {
+            kyberNetwork = _kyberNetwork;
+            emit KyberNetworkUpdated(kyberNetwork);
+        }
+    }
+
+    /// @dev Allow to set kyberNetworkProxy address by daoOperator
+    /// @param _newProxy new kyberNetworkProxy contract
+    function setKyberProxy(IKyberProxy _newProxy) external onlyDaoOperator {
+        require(_newProxy != IKyberProxy(0), "kyberNetworkProxy 0");
```

Comment:
actually this code is duplicated.
right?

so we can inherit kyberFeeHandler and only over run functions that are different??